# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/PeerConnection.jsm
# Commit: 0b03dd9d20ac
# Full Hash: 0b03dd9d20ace235502714101097fe34d1864563
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 1
# Description:
#   Bug 1493613 - Update muted state through MediaStreamTrackSource. r=bwc,smaug
#   
#   This ensures all clones of the original track also receives the new muted state.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37933
# ==============================================================================

diff -r cf84eeffed22 -r 0b03dd9d20ac dom/media/PeerConnection.jsm
--- a/dom/media/PeerConnection.jsm	Tue Jul 30 14:55:48 2019 +0000
+++ b/dom/media/PeerConnection.jsm	Wed Jul 31 09:35:06 2019 +0000
@@ -1576,23 +1576,21 @@
   _processTrackAdditionsAndRemovals() {
     const removeList = [];
     const addList = [];
-    const muteTracks = [];
+    const muteTransceiverReceiveTracks = [];
     const trackEventInits = [];
 
     for (const transceiver of this._transceivers) {
       transceiver.receiver.processTrackAdditionsAndRemovals(transceiver, {
         removeList,
         addList,
-        muteTracks,
+        muteTransceiverReceiveTracks,
         trackEventInits,
       });
     }
 
-    muteTracks.forEach(track => {
+    muteTransceiverReceiveTracks.forEach(transceiver => {
       // Check this as late as possible, in case JS has messed with this state.
-      if (!track.muted) {
-        track.mutedChanged(true);
-      }
+      transceiver.setReceiveTrackMuted(true);
     });
 
     for (const { stream, track } of removeList) {
@@ -2552,7 +2550,7 @@
 
   processTrackAdditionsAndRemovals(
     transceiver,
-    { removeList, addList, muteTracks, trackEventInits }
+    { removeList, addList, muteTransceiverReceiveTracks, trackEventInits }
   ) {
     const receiver = this.__DOM_IMPL__;
     const track = this.track;
@@ -2572,7 +2570,7 @@
         // New track, set in case streamsAdded is empty
         needsTrackEvent = true;
       } else {
-        muteTracks.push(track);
+        muteTransceiverReceiveTracks.push(this._transceiverImpl);
       }
     }
 